api/src/main/java/com/datastrato/gravitino/authorization/User.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.authorization;
+
+import com.datastrato.gravitino.Auditable;
+import java.util.List;
+import java.util.Map;
+
+/** The interface of a user. The user is the entity which executes every operation. */
+public interface User extends Auditable {
+
+  /**
+   * The name of the user.
+   *
+   * @return The name of the user.
+   */
+  String name();
```

Comment:
What's the difference between name and display name, also you have firstName and lastName?

Reply:
Like other systems, name is an identifier for the entity. Display name can be used for UI display. First name and Last name will used for the user name details. Other systems have similar design.